src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -775,8 +787,10 @@ static void create_rr_locked(grpc_exec_ctx *exec_ctx, glb_lb_policy *glb_policy,
 /* glb_policy->rr_policy may be NULL (initial handover) */
 static void rr_handover_locked(grpc_exec_ctx *exec_ctx,
                                glb_lb_policy *glb_policy) {
-  GPR_ASSERT(glb_policy->serverlist != NULL &&
-             glb_policy->serverlist->num_servers > 0);
+  GPR_ASSERT((glb_policy->fallback && glb_policy->fallback_addresses != NULL &&
```

Comment:
This assert is now too long. If it fails, which clause is the invalid one? This could have been argued even for the previous contents, but at least then they were both about `glb_policy->serverlist`. 

In general, not just for asserts, I find that overly composed conditions are difficult to read. I usually create const bools with descriptive variable names that, if I really have to, then compose. That way at least I can read it as if it were natural language. But again, in this case, the main issue is that the assert granularity is too coarse.